File: consumer/Replicant/python/src/founders.py
try:
    from .capsule import Capsule
    from .agent import Agent, Traits
    from .leighton import LambdaState
except ImportError:
    from capsule import Capsule
    from agent import Agent, Traits
    from leighton import LambdaState


FOUNDER_MANIFEST = [
    {
        "name": "Sagan",
        "role": "Founder",
        "traits": {"forage_bias": 0.55, "deposit_rate": 0.40,
                   "scepticism": 0.30, "broadcast_cost": 0.50},
        "lambda_0": 1.20,
        "birth_tick": 0,
        "birth_pos": (50.0, 50.0),
        "inherits": ["replicant/protocol/run-v1"]
    },
    {
        "name": "Dyson",
        "role": "Scout",
        "traits": {"forage_bias": 0.80, "deposit_rate": 0.10,
                   "scepticism": 0.20, "broadcast_cost": 0.30},
        "lambda_0": 1.00,
        "birth_tick": 5,
        "birth_pos": (48.0, 52.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Lovelace",
        "role": "Builder",
        "traits": {"forage_bias": 0.20, "deposit_rate": 0.90,
                   "scepticism": 0.40, "broadcast_cost": 0.20},
        "lambda_0": 0.90,
        "birth_tick": 12,
        "birth_pos": (52.0, 48.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Turing",
        "role": "Attester",
        "traits": {"forage_bias": 0.30, "deposit_rate": 0.10,
                   "scepticism": 0.95, "broadcast_cost": 0.10},
        "lambda_0": 1.10,
        "birth_tick": 18,
        "birth_pos": (45.0, 55.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Curie",
        "role": "Forager",
        "traits": {"forage_bias": 0.70, "deposit_rate": 0.80,
                   "scepticism": 0.05, "broadcast_cost": 0.60},
        "lambda_0": 0.80,
        "birth_tick": 25,
        "birth_pos": (55.0, 45.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Newton",
        "role": "Broadcaster",
        "traits": {"forage_bias": 0.10, "deposit_rate": 0.30,
                   "scepticism": 0.60, "broadcast_cost": 0.90},
        "lambda_0": 1.05,
        "birth_tick": 30,
        "birth_pos": (50.0, 50.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Tesla",
        "role": "Explorer",
        "traits": {"forage_bias": 0.95, "deposit_rate": 0.00,
                   "scepticism": 0.10, "broadcast_cost": 0.00},
        "lambda_0": 0.70,
        "birth_tick": 42,
        "birth_pos": (40.0, 60.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Pasteur",
        "role": "Healer",
        "traits": {"forage_bias": 0.10, "deposit_rate": 0.70,
                   "scepticism": 0.80, "broadcast_cost": 0.10},
        "lambda_0": 1.30,
        "birth_tick": 55,
        "birth_pos": (50.0, 50.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Shannon",
        "role": "Signal",
        "traits": {"forage_bias": 0.40, "deposit_rate": 0.20,
                   "scepticism": 0.70, "broadcast_cost": 0.40},
        "lambda_0": 0.95,
        "birth_tick": 67,
        "birth_pos": (52.0, 52.0),
        "inherits": ["replicant/agent/founder-sagan"]
    },
    {
        "name": "Darwin",
        "role": "Observer",
        "traits": None,
        "lambda_0": 2.00,
        "birth_tick": 100,
        "birth_pos": (50.0, 50.0),
        "inherits": ["replicant/agent/founder-sagan"]
    }
]


def create_founders(key=None):
    """Instantiate all 10 founders from manifest."""
    agents = {}

    for manifest in FOUNDER_MANIFEST:
        name = manifest["name"]

        capsule = Capsule.mint(
            inherits=manifest["inherits"],
            declaration={
                "name": name,
                "role": manifest["role"],
                "traits": manifest["traits"],
                "birth_tick": manifest["birth_tick"],
                "birth_pos": manifest["birth_pos"],
                "lambda_0": manifest["lambda_0"]
            },
            licence="MSL-1.0",
            key=key
        )

        if manifest["traits"] is not None:
            traits = Traits(**manifest["traits"])
        else:
            traits = None

        # New LambdaState: base = initial λ
        lambda_state = LambdaState(base=manifest["lambda_0"])

        agent = Agent(
            scp_id=capsule.scp_id,
            capsule=capsule,
            x=manifest["birth_pos"][0],
            y=manifest["birth_pos"][1],
            traits=traits,
            lambda_state=lambda_state,
            birth_tick=manifest["birth_tick"],
            role=manifest["role"]
        )

        agents[name] = agent

    return agents

File: consumer/Replicant/rust/src/founders.rs
//! Founders module - creates the initial set of agents

use crate::core::*;
use crate::agent::*;
use rand::Rng;
use std::collections::HashMap;

pub fn create_founders() -> HashMap<String, Agent> {
    let mut founders = HashMap::new();
    let mut rng = rand::thread_rng();

    let sagan_capsule = Capsule::mint(
        vec!["replicant/protocol/run-v1".to_string()],
        serde_json::json!({"name": "Sagan"})
    );
    let sagan = Agent::new(
        sagan_capsule.scp_id.clone(),
        sagan_capsule,
        50.0, 50.0,
        Traits { forage_bias: 0.6, deposit_rate: 0.7, scepticism: 0.4, broadcast_cost: 0.6 },
        LambdaState::new(1.2),
        Role::Founder,
        // ~80% Generalist, ~20% Purist.
        // This replaces the old `is_specialist` boolean.
        if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist },
        0,
    );
    founders.insert("Sagan".to_string(), sagan);

    let dyson_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Dyson"}));
    let dyson = Agent::new(
        dyson_capsule.scp_id.clone(),
        dyson_capsule,
        45.0, 55.0,
        Traits { forage_bias: 0.8, deposit_rate: 0.3, scepticism: 0.6, broadcast_cost: 0.4 },
        LambdaState::new(1.0),
        Role::Scout,
        if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist },
        0,
    );
    founders.insert("Dyson".to_string(), dyson);

    let lovelace_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Lovelace"}));
    let lovelace = Agent::new(
        lovelace_capsule.scp_id.clone(),
        lovelace_capsule,
        55.0, 45.0,
        Traits { forage_bias: 0.4, deposit_rate: 0.8, scepticism: 0.5, broadcast_cost: 0.7 },
        LambdaState::new(0.9),
        Role::Builder,
        if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist },
        0,
    );
    founders.insert("Lovelace".to_string(), lovelace);

    let turing_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Turing"}));
    let turing = Agent::new(
        turing_capsule.scp_id.clone(),
        turing_capsule,
        52.0, 52.0,
        Traits { forage_bias: 0.3, deposit_rate: 0.2, scepticism: 0.9, broadcast_cost: 0.3 },
        LambdaState::new(1.1),
        Role::Attester,
        if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist },
        0,
    );
    founders.insert("Turing".to_string(), turing);

    let curie_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Curie"}));
    let curie = Agent::new(
        curie_capsule.scp_id.clone(),
        curie_capsule,
        48.0, 48.0,
        Traits { forage_bias: 0.9, deposit_rate: 0.6, scepticism: 0.3, broadcast_cost: 0.2 },
        LambdaState::new(0.8),
        Role::Forager,
        if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist },
        0,
    );
    founders.insert("Curie".to_string(), curie);

    let newton_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Newton"}));
    let newton = Agent::new(newton_capsule.scp_id.clone(), newton_capsule, 60.0, 60.0, Traits::default(), LambdaState::default(), Role::Broadcaster, if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist }, 0);
    founders.insert("Newton".to_string(), newton);

    let tesla_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Tesla"}));
    let tesla = Agent::new(tesla_capsule.scp_id.clone(), tesla_capsule, 40.0, 40.0, Traits::default(), LambdaState::default(), Role::Explorer, if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist }, 0);
    founders.insert("Tesla".to_string(), tesla);

    let pasteur_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Pasteur"}));
    let pasteur = Agent::new(pasteur_capsule.scp_id.clone(), pasteur_capsule, 35.0, 65.0, Traits::default(), LambdaState::default(), Role::Healer, if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist }, 0);
    founders.insert("Pasteur".to_string(), pasteur);

    let shannon_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Shannon"}));
    let shannon = Agent::new(shannon_capsule.scp_id.clone(), shannon_capsule, 65.0, 35.0, Traits::default(), LambdaState::default(), Role::Signal, if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist }, 0);
    founders.insert("Shannon".to_string(), shannon);

    let darwin_capsule = Capsule::mint(vec![], serde_json::json!({"name": "Darwin"}));
    let darwin = Agent::new(darwin_capsule.scp_id.clone(), darwin_capsule, 50.0, 50.0, Traits::default(), LambdaState::default(), Role::Observer, if rng.gen_bool(0.8) { Archetype::Generalist } else { Archetype::Purist }, 0);
    founders.insert("Darwin".to_string(), darwin);

    founders
}